update(api.HTMLElement.tabIndex): update Edge and IE compatibility

CITguy opened the pull request:
Edge and IE have a buggy implementation of the `tabIndex` IDL attribute.

https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/4365703/

saschanaz commented:
This has been fixed in Windows 1809 update, not sure about earlier versions.

CITguy commented:
Should we hold off on this until Windows 10 (1809) is publicly available to verify the correction?

I wouldn't want to add notes that are incomplete.

Elchi3 commented:
Yes, we can wait merging this until we're able to verify that, if you like.

ddbeck commented:
@CITguy, it looks like 1809 is now publicly available. Does that change the data for this PR?

CITguy commented:
I can verify that Edge 18 has fixed the bug. I'll update this PR to reflect the fix.

CITguy commented:
* I've updated the data to denote partial implementation for IE 7+ and Edge 13-17. 
* I added an explicit entry for Edge 18 to reflect the bug fix.

Please let me know if I need to simplify.

saschanaz commented on api/HTMLElement.json:
```diff
@@ -1848,20 +1848,36 @@
+                "notes": "Returns incorrect value for elements without an explicit tabindex attribute. See <a href='https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/4365703/'>issue 4365703</a> for details."
+              },
+              {
+                "version_added": "18"
```
AFAIK the newer entry typical goes first.

CITguy replied:
done

saschanaz commented on api/HTMLElement.json:
```diff
@@ -1848,20 +1848,36 @@
+            ],
+            "edge_mobile": [
+              {
+                "version_added": "18"
```
Edge mobile 18 doesn't really exist. See #3116

CITguy replied:
reverted to `"version_added": true`
